ingestion/src/metadata/ingestion/source/database/trino/connection.py

```diff
@@ -77,14 +76,51 @@ def get_connection_url(connection: TrinoConnection) -> str:
 
 @connection_with_options_secrets
 def get_connection_args(connection: TrinoConnection):
+    if not connection.connectionArguments:
+        connection.connectionArguments = init_empty_connection_arguments()
+
     if connection.proxies:
         session = Session()
         session.proxies = connection.proxies
-        if not connection.connectionArguments:
-            connection.connectionArguments = init_empty_connection_arguments()
 
         connection.connectionArguments.__root__["http_session"] = session
 
+    if isinstance(connection.authType, basicAuth.BasicAuth):
+        connection.connectionArguments.__root__["auth"] = BasicAuthentication(
+            connection.username, connection.authType.password.get_secret_value()
+        )
+        connection.connectionArguments.__root__["http_scheme"] = "https"
+
+    elif isinstance(connection.authType, jwtAuth.JwtAuth):
+        connection.connectionArguments.__root__["auth"] = JWTAuthentication(
+            connection.authType.jwt.get_secret_value()
+        )
+        connection.connectionArguments.__root__["http_scheme"] = "https"
+
+    elif isinstance(connection.authType, AzureCredentials):
+        if not connection.authType.scopes:
+            raise ValueError(
+                "Azure Scopes are missing, please refer https://learn.microsoft.com/en-gb/azure/mysql/flexible-server/how-to-azure-ad#2---retrieve-microsoft-entra-access-token and fetch the resource associated with it, for e.g. https://ossrdbms-aad.database.windows.net/.default"
+            )
+
+        azure_client = AzureClient(**connection.authType.dict()).create_client()
+
+        access_token_obj = azure_client.get_token(
+            *connection.authType.scopes.split(",")
+        )
+
+        connection.connectionArguments.__root__["auth"] = JWTAuthentication(
+            access_token_obj.token
+        )
+        connection.connectionArguments.__root__["http_scheme"] = "https"
+
+    elif (
+        connection.authType
+        == noConfigAuthenticationTypes.NoConfigAuthenticationTypes.OAuth2
+    ):
+        connection.connectionArguments.__root__["auth"] = OAuth2Authentication()
+        connection.connectionArguments.__root__["http_scheme"] = "https"
+
```

Comment:
@mgorsk1 correct me if I'm wrong, but this method would only work if you are running a trino metadata job on your local machine via CLI mode where you have access to a browser, as `OAuth2Authentication` would print a link or redirect you to a the same link where you can complete the authentication

Reply:
that's correct, I added id because that's the scenario we needed. The example scenario is:
- trino configured with oauth2
- users run profiling jobs using OM SDK from their private environments
- profiling jobs takes time so we need a mechanism that will collect access_token but also refresh it